### PR TITLE
add missing organization filter on slides query

### PR DESCRIPTION
**Pull request by nicolas43000**

### CODE REVIEW

**Review by chrisaligent — APPROVED** [review]
Good catch on this one, exactly how we would have implemented this fix :+1:

File: src/Layout/DataProvider/SliderDataProvider.php [hunk 1]
```diff
@@ -24,50 +23,58 @@ class SliderDataProvider
...
-        
-        /** @var Slider $slider */
-        $slider = $this->sliderRepository->find($sliderId);
-        
-        return $slider->getCode();
+        $slider = $this->getSliderUsedConfig();
+        if (!is_null($slider)) {
+            return $slider->getCode();
+        }
+        return;
```
Comment (gplanchat) [inline src/Layout/DataProvider/SliderDataProvider.php:78]: Please, could you explicitly return a `null` value ?

File: src/Layout/DataProvider/SliderDataProvider.php [hunk 2]
```diff
@@ -24,50 +23,58 @@ class SliderDataProvider
...
 
     public function getSliderUsedConfig()
     {
         /** @var Slider $slider */
         $sliderId = $this->config->get('slider_bundle.slider');
-        return  $this->sliderRepository->find($sliderId);
+        if (!is_null($sliderId)) {
+            return $this->sliderRepository->find($sliderId);
+        }
+        return;
```
Comment (gplanchat) [inline src/Layout/DataProvider/SliderDataProvider.php:69]: Please, could you explicitly return a `null` value ?

File: src/Layout/DataProvider/SliderDataProvider.php [hunk 3]
```diff
@@ -24,50 +23,58 @@ class SliderDataProvider
...
         $context = $criteria->toArray();
-        $slides = $this->sliderRepository->getSlidesBySliderCode($sliderCode);
-        $slidesVisibles = $this->slideResolver->getVisibleSlides($slides,$context);
-        return $slidesVisibles;
+        $slides = $this->sliderRepository->getSlidesBySliderCode($sliderCode, $this->tokenAccessor->getOrganization());
+        $slidesVisibles = $this->slideResolver->getVisibleSlides($slides, $context);
+        if (!is_null($slidesVisibles)) {
+            return $slidesVisibles;
+        }
+        return;
```
Comment (gplanchat) [inline src/Layout/DataProvider/SliderDataProvider.php:59]: Please, could you explicitly return an empty array `[]` value ?

**Review by gplanchat — APPROVED** [review]

**Pull request merged**